src/win/rlimit.c

```diff
@@ -129,99 +118,79 @@ int setrlimit(int resource, const struct rlimit *rlp)
   if(rlp->rlim_cur < rlimits[resource].rlim_max)
     rlimits[resource].rlim_cur = rlp->rlim_cur;
   else
-    iRet = EINVAL;
+    return EINVAL;
   //
   // We should not let the user set the max value.  However,
   // since currently there is no defined source for initial
   // values, we will let the user change the max value.
   //
   rlimits[resource].rlim_max = rlp->rlim_max;
 
-  return iRet;
+  return 0;  // success
 }
 
 /////////////////////////////////////////////////////////////////
 // Wrap the real fwrite() with this rfwrite() function, which is
 // resource limit aware.
 //
-//
 ///////////////////////////////////////////////////////////////size_t
 size_t rfwrite(const void *buffer, size_t size, size_t count, FILE *stream)
 {
-  long position;
-  size_t written;
-  __int64 liByteCount, liPosition;
   //
   // Convert the count to a large integer (64 bit integer)
   //
-  liByteCount = (__int64)count;
+  __int64 liByteCount = (__int64)count;
 
   //
   // Get the current file position
   //
-  position = ftell(stream);
-  liPosition = (__int64)position;
+  __int64 liPosition = (__int64)ftell(stream);
 
   //
   // Check to make sure the write will not exceed the RLIMIT_FSIZE limit.
   //
-  if((liPosition + liByteCount) > rlimits[RLIMIT_FSIZE].rlim_cur)
-  {
+  if(liPosition + liByteCount > rlimits[RLIMIT_FSIZE].rlim_cur)
     //
     // report an error
     //
-    written = 0;
-  }
-  else
-  {
-    //
-    // Do the actual write the user requested
-    //
-    written = fwrite(buffer, size, count, stream);
-  }
-  return written;
+    return 0;
+
+  //
+  // Do the actual write the user requested
+  //
+  return fwrite(buffer, size, count, stream);
 }
 
 /////////////////////////////////////////////////////////////////
 // Wrap the real _write() function with the _rwrite() function
 // which is resource aware.
 //
-//
 ///////////////////////////////////////////////////////////////int
 int _rwrite(int handle, const void *buffer, unsigned int count)
 {
-  long position;
-  DWORD dwWritten;
-  int64_t liByteCount, liPosition;
   //
   // Convert the count to a large integer
   //
-  liByteCount = (__int64)count;
+  int64_t liByteCount = (__int64)count;
 
   //
   // Get the Current file position
   //
-  position = _tell(handle);
-  liPosition = (__int64)position;
+  int64_t liPosition = (__int64)_tell(handle);
 
   //
   // Check to make sure the write will not exceed the RLIMIT_FSIZE limit.
   //
-  if((liPosition + liByteCount) > rlimits[RLIMIT_FSIZE].rlim_cur)
-  {
+  if(liPosition + liByteCount > rlimits[RLIMIT_FSIZE].rlim_cur)
     //
     // report an error
     //
-    dwWritten = 0;
-  }
-  else
-  {
-    //
-    // Do the actual write the user requested
-    //
-    dwWritten = _write(handle, buffer, count);
-  }
-  return dwWritten;
+    return = 0;
```

Comment:
`return = 0` happens ;) this is the one CI complains about ;)

Reply:
This code is Windows specific, but I am on Linux. That's why I was not able to compile it for myself. Luckily, we have an automated build chain for all architectures.